### ISSUE TITLE
Unable to publish using latest version of @aws-amplify/cli (4.42.0)

### ISSUE THREAD

**Issue by SimonHoiberg** [ambientlight/amplify-cli-action#16]
In the latest version of @aws-amplify/cli (4.42.0), an additional step was added before finally publishing files to S3.
GitHub actions get stuck here and report a false positive.
(_See code snippet below_)

The files are never being uploaded to S3.

Downgrading to __@aws-amplify/cli@4.41.2__ using the flag `amplify_cli_version: '4.41.2'` solves the issue.

```console
frontend build command exited with code 0
Publish started for S3AndCloudFront
? Select the authentication method you want to use: (Use arrow keys)
❯ AWS profile 
  AWS access keys 
```

**Comment by sar** [comment]
Just hit this issue on some production endpoints.

Downgrading to `@aws-amplify/cli@4.41.2` throws compatibility errors with newly published stacks.

```log
Fri, 19 Feb 2021 04:08:14 GMT Invalid feature flag configuration
Fri, 19 Feb 2021 04:08:14 GMT
Fri, 19 Feb 2021 04:08:14 GMT These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:
Fri, 19 Feb 2021 04:08:14 GMT   - appsync
Fri, 19 Feb 2021 04:08:14 GMT   - graphqltransformer.secondarykeyasgsi
Fri, 19 Feb 2021 04:08:14 GMT
Fri, 19 Feb 2021 04:08:14 GMT This issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.
Fri, 19 Feb 2021 04:08:14 GMT
Fri, 19 Feb 2021 04:08:14 GMT Ensure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.
Fri, 19 Feb 2021 04:08:14 GMT
Fri, 19 Feb 2021 04:08:14 GMT Learn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags
Fri, 19 Feb 2021 04:08:16 GMT found existing environment prod
Fri, 19 Feb 2021 04:08:17 GMT Invalid feature flag configuration
Fri, 19 Feb 2021 04:08:17 GMT
Fri, 19 Feb 2021 04:08:17 GMT These feature flags are defined in the "amplify/cli.json" configuration file and are unknown to the currently running Amplify CLI:
Fri, 19 Feb 2021 04:08:17 GMT   - appsync
Fri, 19 Feb 2021 04:08:17 GMT   - graphqltransformer.secondarykeyasgsi
Fri, 19 Feb 2021 04:08:17 GMT
Fri, 19 Feb 2021 04:08:17 GMT This issue likely happens when the project has been pushed with a newer version of Amplify CLI, try updating to a newer version.
Fri, 19 Feb 2021 04:08:17 GMT
Fri, 19 Feb 2021 04:08:17 GMT Ensure that the CI/CD pipeline is not using an older or pinned down version of Amplify CLI.
Fri, 19 Feb 2021 04:08:17 GMT
Fri, 19 Feb 2021 04:08:17 GMT Learn more about feature flags: https://docs.amplify.aws/cli/reference/feature-flags
```

**Comment by amcdnl** [comment]
Also hitting this issue - its not related to this project though.

**Comment by SimonHoiberg** [comment]
@sar Yeah, this is a bit painful.
Best recommendation is to wipe the CloudFormation stack and do `amplify init` with the downgraded version.

@amcdnl This is not really an issue with the amplify repository. They added an extra step for a more user-friendly experience.
It's this project that needs to update to accommodate the new changes.

**Comment by sar** [comment]
> @amcdnl This is not really an issue with the amplify repository. They added an extra step for a more user-friendly experience.

Does `4.44.0` implement `<subcommand>` or support passing flag `--auth <type>` on  `amplify push|pull|publish`? 

This can be added as required `arg` or coalesce to default env on `entrypoint.sh` in a quick pr:

https://github.com/ambientlight/amplify-cli-action/blob/60165d38e083b3e52d7d026e038bdfb08a2fad14/entrypoint.sh#L49-L55

**Comment by jackvhall** [comment]
Has there been any progress on this issue? What is the current workaround if not able to downgrade the CLI version and no --auth flag available?

**Comment by tobocop2** [comment]
I've made a fix here, and am able to successfully publish with the latest amplify now:

#23

**Comment by ambientlight** [comment]
Closed by #23 and available in [0.2.2](https://github.com/ambientlight/amplify-cli-action/releases/tag/0.2.2)
